Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -0,0 +1,52 @@
+import SwiftSyntax
+
+/// Rule to require that any state properties in SwiftUI be declared as private.
+/// State properties should only be accessible from inside the View's body, or from methods called by it
+struct PrivateSwiftUIStatePropertyRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRule {
+    var configuration = SeverityConfiguration<Self>(.warning)
+
+    static let description = RuleDescription(
+        identifier: "private_swiftui_state",
+        name: "Private SwiftUI @State Properties",
+        description: "SwiftUI's state properties should be private",
+        kind: .lint,
+        nonTriggeringExamples: PrivateSwiftUIStatePropertyRuleExamples.nonTriggeringExamples,
```

Comment:
If a rule's implementation is short and the number of examples is rather small, we typically inline the examples.

Reply:
👍 I wasn't certain on the convention here. I can revert the change, it's an easy one!